content/blog/zephyr-how-to-use-psram/index.md

```diff
@@ -0,0 +1,605 @@
+I (97) flash_init: SPI Mode       : DIO
+I (100) flash_init: SPI Flash Size : 8MB
+I (104) soc_random: Enabling RNG early entropy source
+I (109) boot: DRAM: lma 0x00000020 vma 0x3fc8e988 len 0x1c2c   (7212)
+I (115) boot: IRAM: lma 0x00001c54 vma 0x40374000 len 0xa968   (43368)
+I (121) boot: padd: lma 0x0000c5c8 vma 0x00000000 len 0x3a30   (14896)
+I (127) boot: IMAP: lma 0x00010000 vma 0x42000000 len 0x3e48   (15944)
+I (134) boot: padd: lma 0x00013e50 vma 0x00000000 len 0xc1a8   (49576)
+I (140) boot: DMAP: lma 0x00020000 vma 0x3c010000 len 0x12ec   (4844)
+I (146) boot: Image with 6 segments
+I (149) boot: DROM segment: paddr=00020000h, vaddr=3c010000h, size=012F0h (  4848) map
+I (157) boot: IROM segment: paddr=00010000h, vaddr=42000000h, size=03E46h ( 15942) map
+I (176) soc_random: Disabling RNG early entropy source
+I (176) boot: Disabling glitch detection
+I (176) boot: Jumping to the main image...
+I (177) octal_psram: vendor id    : 0x0d (AP)
+I (181) octal_psram: dev id       : 0x02 (generation 3)
+I (186) octal_psram: density      : 0x03 (64 Mbit)
+I (190) octal_psram: good-die     : 0x01 (Pass)
+I (195) octal_psram: Latency      : 0x01 (Fixed)
+I (199) octal_psram: VCC          : 0x01 (3V)
+I (203) octal_psram: SRF          : 0x01 (Fast Refresh)
+I (208) octal_psram: BurstType    : 0x01 (Hybrid Wrap)
+I (213) octal_psram: BurstLen     : 0x01 (32 Byte)
+I (217) octal_psram: Readlatency  : 0x02 (10 cycles@Fixed)
+I (223) octal_psram: DriveStrength: 0x00 (1/1)
+I (227) MSPI Timing: PSRAM timing tuning index: 6
+I (231) esp_psram: Found 8MB PSRAM device
+I (235) esp_psram: Speed: 80MHz
+I (245) mmu_psram: Instructions copied and mapped to SPIRAM
+I (252) mmu_psram: Read only data copied and mapped to SPIRAM
+I (652) esp_psram: SPI SRAM memory test OK
+I (708) heap_runtime: ESP heap runtime init at 0x3fc93030 size 345 kB.
+
+*** Booting Zephyr OS build v4.0.0-1471-ge60f04096cd8 ***
+Hello World! esp32s3_devkitc/esp32s3/procpu
+
+```
+
+Here, we can confirm that instructions and read-only data were copied and mapped to PSRAM, which can optimize RAM usage when there is code that needs to execute during an SPI1 Flash operation.
+
+## 9. Conclusion
+
+Throughout this article, we explored three different strategies for utilizing PSRAM: using it as a heap for dynamic memory allocation, integrating it as a section in the memory map, and moving code from FLASH to PSRAM. In addition to these three strategies, it is also possible to leverage PSRAM to execute external code compiled as Position Independent Code (PIC), which can be placed in PSRAM by a loader for execution. Another potential use of PSRAM to reduce SRAM usage is by enabling **CONFIG_ESP32_WIFI_NET_ALLOC_SPIRAM**, allowing the ESP32 Wi-Fi stack to dynamically allocate memory from PSRAM. However, as of the time of writing, there is an unfixed bug preventing applications built with **CONFIG_ESP32_WIFI_NET_ALLOC_SPIRAM** from functioning properly.
+
+## 10. References
```

Comment:
For the references, please add the links using the Markdown format:

`[Espressif Official Site](https://www.espressif.com/en)`